### PR TITLE
fix install with pip10

### PR DESCRIPTION
**Pull request by mduret**
hello,

I propose a simple fix for this issue : https://github.com/spiral-project/ihatemoney/issues/340

Have a good day

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -2,16 +2,24 @@
...
-# Get requirements from the requirements.txt file.
-pip_requirements = parse_requirements("requirements.txt", session=PipSession())
-install_requires = [str(ir.req) for ir in pip_requirements]
+
+import pip
+if pip.__version__ < "10":
+    try:
+        from pip.req import parse_requirements
+        from pip.download import PipSession
+    except ImportError:
```
Comment (almet) [inline setup.py]: I think the try/except for the import should be just next to "import pip" rather than here.

File: setup.py [hunk 2]
```diff
@@ -2,16 +2,24 @@
...
+        print('Cannot find pip.')
+        raise
+    # Get requirements from the requirements.txt file.
+    pip_requirements = parse_requirements("requirements.txt", session=PipSession())
+    install_requires = [str(ir.req) for ir in pip_requirements]
+else:
+    def parse_requirements(filename):
+        """ load requirements from a pip requirements file """
+        lineiter = (line.strip() for line in open(filename))
+        return [line for line in lineiter if line and not line.startswith("#")]
```
Comment (almet) [inline setup.py]: If we have this utility, I think we should use it in all occasions, and throw away the use of pip's version. That's less differences and less trouble maintaining two versions.

Comment (almet) [reply setup.py]: Could we have a bunch of tests for this new function?

File: setup.py [hunk 3]
```diff
@@ -2,16 +2,24 @@
...
+    except ImportError:
+        print('Cannot find pip.')
+        raise
+    # Get requirements from the requirements.txt file.
+    pip_requirements = parse_requirements("requirements.txt", session=PipSession())
+    install_requires = [str(ir.req) for ir in pip_requirements]
+else:
+    def parse_requirements(filename):
+        """ load requirements from a pip requirements file """
+        lineiter = (line.strip() for line in open(filename))
```
Comment (almet) [inline setup.py]: We open the file but never close it. (hint: it's handled automatically when using a `with` statement)

**Comment by mduret** [comment]
Thanks for the tricks, I gonna propose a more elegant fix in few days.

**Comment by mduret** [comment]
I propose a more simple way : it works with pip 9 and pip 10.
I hope it can help.

Have a good one,

**Comment by almet** [comment]
Thanks! That seems to be a simpler way to handle this. Could you provide a few tests for the created function? (So we're sure it does what it should?). Thanks!

**Comment by mduret** [comment]
Ok, thx I going to make this test.

**Comment by JocelynDelalande** [comment]
I allowed myself to rename as your fix is broader than just docker.

**Comment by mduret** [comment]
Hi,

I spend some time on this issue but I don't know where to put this test.

I am not an expert but `setup.py` is not part of the package so I do not know how to test it in the package.

Usualy, I put `-e .` in `requirements.txt` and I put requirements in the `setup` attribute `install_requires` of the `setup.py` file.

If you have an idea...

Thanks and have a good day.

**Comment by JocelynDelalande** [comment]
> I am not an expert but setup.py is not part of the package so I do not know how to test it in the package.

Good point. I searched/tested things a bit myself and cannot find a way to do that (test setup.py functions).

py.test seem to fail at parsing it (`py.test setup.py` will crash). 

So unless someone comes quickly with a solution, and this bug (#340) maintains ihm in a quite broken state, I'd suggest merging as-is.

**Comment by almet** [comment]
Yes, after some look around, I don't see any obvious way to do it as well. Let's merge! Thanks for your work and patience here :)

**Pull request merged**